### PR TITLE
[WIP] Save votes in local storage

### PR DESCRIPTION
**Pull request by TimNN**
Part of a potential solution to #715.

This uses the proposed sisyphus library.

While user information is currently preserved across reloads and navigation actions, user data is automatically cleared when the form is submitted.

Although this can be trivially disabled, this would lead to user data never being deleted from local storage (which is very undesirable I think).

Potential solutions:
* clear local storage on confirmation page load **Problem:** the user is redirect to index, making it non-trivial to pass the required data to the confirmation page
* submit the form using ajax and only clear the form upon success

I believe that the second solution would be preferable.

Currently only tested on Chrome on OS X.

### CODE REVIEW

**Comment by janno42** [comment]
yes, please go for the ajax post request and clean on success.

File: evap/student/templates/student_vote.html [hunk 1]
```diff
@@ -101,11 +104,29 @@
...
+            sisyphus.saveAllData()
         }
         function selectedAnswer(fieldName) {
             // remove error highlighting when an answer was selected
             $(".choice-error[name=" + fieldName + "]").removeClass("choice-error");
         }
         preventAccidentalClosing = true;
+
+        $(function() {
+            sisyphus = $("#student-vote-form").sisyphus({
```
Comment (janno42) [inline evap/student/templates/student_vote.html]: can you replace the variable definition in line 93 above with `$("#student-vote-form").sisyphus({...});` and remove the rest?

File: evap/student/templates/student_vote.html [hunk 2]
```diff
@@ -101,11 +104,29 @@
...
+                locationBased: true,
+                excludeFields: $("input[type=hidden]"), // exclude the csrf token
+                onRestore: function () {
+                    // If the form contains errors, is submitted, then changes are made and the page is reloaded,
+                    // some labels will have been set active by the server but the underlying checkbox is no longer
+                    // checked because it has been updated by sisyphus
```
Comment (janno42) [inline evap/student/templates/student_vote.html]: these comments were not perfectly understandable to me at first. can you please just write a more general `update vote button states` before the two lines of code?

**Comment by TimNN** [comment]
I currently don't have time to continue working on this, especially since using ajax for submission will require a non-trivial amount of work as all the error display would have to be handled in JS as well.

**Comment by janno42** [comment]
@TimNN allright, then i'll take over. thanks so far :)

**Comment by janno42** [comment]
replaced by #760

**Pull request closed without merging**